### PR TITLE
 Update PHP snippet generator based on the compile check.

### PR DESCRIPTION
**Pull request by shinfan**
- Fix errors and corner cases found from PHP compile check.
- Add $postBody if the method contains a request parameter.
- Update baseline tests.

### CODE REVIEW

File: src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java [hunk 1]
```diff
@@ -52,13 +54,31 @@
...
+          // As a work-around in Javascript, Disovery importor treats number format strings
+          // as TYPE_INT64 or TYPE_UINT64 depending on the string format.
+          .put(Field.Kind.TYPE_INT64, "\'0\'")
+          .put(Field.Kind.TYPE_UINT64, "\'0\'")
           .build();
 
   /**
+   * A set that contains the method names that have extra suffix in the PHP client code.
+   */
+  private static final ImmutableSet<String> RENAMED_METHODS = ImmutableSet.<String>builder()
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java]: Give an example here. What do we need to rename it to? How is it different from others?

File: src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java [hunk 2]
```diff
@@ -74,23 +94,55 @@ public void resetState(PhpSnippetSet<?> phpSnippetSet, PhpContextCommon phpCommo
...
+   */
   public String getMethodName(Method method) {
     String name = super.getMethodName(method);
-    List<String> resources = getApiaryConfig().getResources(method.getName());
-    if (resources != null && resources.size() > 0) {
-      name = name + lowerCamelToUpperCamel(resources.get(resources.size() -1));
+    if (RENAMED_METHODS.contains(name)) {
+      List<String> resources = getApiaryConfig().getResources(method.getName());
+      for (int i = 0; i < resources.size(); i++) {
+        name = name + lowerCamelToUpperCamel(resources.get(i));
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java]: Consider using `for (String resource : getResources())`

Consider using StringBuilder.

File: src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java [hunk 3]
```diff
@@ -52,13 +54,31 @@
           .put(Field.Kind.TYPE_BOOL, "false")
           .put(Field.Kind.TYPE_INT32, "0")
           .put(Field.Kind.TYPE_UINT32, "0")
-          .put(Field.Kind.TYPE_INT64, "0")
-          .put(Field.Kind.TYPE_UINT64, "0")
           .put(Field.Kind.TYPE_FLOAT, "0.0")
           .put(Field.Kind.TYPE_DOUBLE, "0.0")
+          // As a work-around in Javascript, Disovery importor treats number format strings
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java]: s/importor/importer/

Could you clarify what you mean by "As a workaround in Javscript"?

File: src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java [hunk 4]
```diff
@@ -52,13 +54,31 @@
...
   /**
+   * A set that contains the method names that have extra suffix in the PHP client code.
+   */
+  private static final ImmutableSet<String> RENAMED_METHODS = ImmutableSet.<String>builder()
+      .add("list")
+      .add("clone")
+      .build();
+
+  /**
+   * A map that contains the requests that require renaming in the PHP client code.
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java]: nit: suggest being explicit about the keys and values.

**Comment by vchudnov-g** [comment]
lgtm but also respond to comments by the others

File: src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java (hunk 3 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java]: Done

File: src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java (hunk 1 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java]: Done.

File: src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java (hunk 4 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java]: Done

File: src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java (hunk 2 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java]: Done

**Comment by shinfan** [comment]
Updated. PTAL.

File: src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java [hunk 5]
```diff
@@ -98,14 +101,14 @@ public void resetState(PhpSnippetSet<?> phpSnippetSet, PhpContextCommon phpCommo
    * Returns the method names used in the PHP client code.
    */
   public String getMethodName(Method method) {
-    String name = super.getMethodName(method);
-    if (RENAMED_METHODS.contains(name)) {
+    StringBuilder builder = new StringBuilder(super.getMethodName(method));
+    if (RENAMED_METHODS.contains(builder.toString())) {
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java]: It feels strange to call toString here.
`contains(super.getMethodName(method))` ?

**Comment by pongad** [comment]
One nit: otherwise LGTM.

**Comment by shinfan** [comment]
Fixed the nit. I will go ahead and merge once all check has passed.
